src/FixedDataTableCellGroup.js

```diff
@@ -119,66 +120,105 @@ class FixedDataTableCellGroupImpl extends React.Component {
   constructor(props) {
     super(props);
     this._initialRender = true;
+    this._staticCells=[];
   }
 
   componentDidMount() {
     this._initialRender = false;
   }
+  shouldComponentUpdate() /*boolean*/ {
+    // Don't add PureRenderMixin to this component please.
+    return true;
+  }
 
   /**
    * Returns Object consisting of keys and widths of the columns in the current cell group.
    * @returns {{keys: [], widths: []}}
    */
   getCellGroupWidth = () => {
+  
     const { columns } = this.props;
     const cellGroupColumnWidths = {
       keys: [],
       widths: [],
     };
+    var colsToRender=this.props.colsToRender || [];
+   
     if (this.props.isHeader) {
-      for (let i = 0, j = columns.length; i < j; i++) {
-        const key = columns[i].props.columnKey || 'cell_' + i;
+      for (let i = 0, j = colsToRender.length; i < j; i++) {
+        var idx=colsToRender[i];
+        
+        if (idx === undefined) {
+          idx =
+            this._staticCells[i] && this._staticCells[i].props.index;
+    
+        }
+        
+        const key = columns[idx].props.columnKey || 'cell_' + idx;
         cellGroupColumnWidths.keys.push(key);
-        cellGroupColumnWidths.widths.push(columns[i].props.width);
+        cellGroupColumnWidths.widths.push(columns[idx].props.width);
       }
     }
     return cellGroupColumnWidths;
   };
+  componentWillUnmount() {
+    this._staticCells.length = 0;
+  }
+
 
   render() /*object*/ {
     var props = this.props;
     var columns = props.columns;
-    var cells = new Array(columns.length);
-    var contentWidth = sumPropWidths(columns);
-
+    var colsToRender=props.colsToRender || [];
+    
+    var isScrolling=props.isScrolling;
+   
+    var contentWidth = this.props.contentWidth;
+    if (isScrolling) {
+      // allow static array to grow while scrolling
+      this._staticCells.length = Math.max(
+        this._staticCells.length,
+        colsToRender.length
+      );
+    } else {
+      // when scrolling is done, static array can shrink to fit the buffer
+      this._staticCells.length = colsToRender.length;
+    }
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
-      var columnProps = columns[i].props;
-      var cellTemplate = columns[i].template;
-
-      var recyclable = _.get(
+    for (var i = 0, j = colsToRender.length; i < j; i++) {
+      var idx=colsToRender[i];
+     
+      if (idx === undefined) {
+        idx =
+          this._staticCells[i] && this._staticCells[i].props.index;
+   
+      }
+   
+    if(!(this.props.columns[idx]))continue;
+      //currentPosition=this.props.colOffsets[idx];
```

Comment:
nit: For consistency with other parts of the codebase, use `colOffsets` rather than `columns[idx].offset`